### ISSUE TITLE
OpenWrt release

### ISSUE THREAD

**Issue by rapiz1** [rapiz1/rathole#133]
Follows https://github.com/rapiz1/rathole/issues/114

It will be useful if the release can be made using OpenWrt SDK and directly .ipk files. This can be done in a separate packaging repository.

**Comment by luojia65** [comment]
Any progress here? :)

**Comment by rapiz1** [comment]
Nope. I'm not using OpenWrt currently and don't know how to package Rust program on that platform. Also it may be hard to distribute binary since the OpenWrt is so diverse. What do you think? Are you interested in setting up such repo?



---Original---
From: "Luo ***@***.***&gt;
Date: Mon, Aug 15, 2022 15:05 PM
To: ***@***.***&gt;;
Cc: "Yujia ***@***.******@***.***&gt;;
Subject: Re: [rapiz1/rathole] OpenWrt release (Issue #133)




 
Any progress here? :)

**Comment by rapiz1** [comment]
Also openssl is an issue. However we can start with not delivering tls feature&nbsp;

---Original---
From: "Luo ***@***.***&gt;
Date: Mon, Aug 15, 2022 15:05 PM
To: ***@***.***&gt;;
Cc: "Yujia ***@***.******@***.***&gt;;
Subject: Re: [rapiz1/rathole] OpenWrt release (Issue #133)




 
Any progress here? :)

**Comment by LouisDK1** [comment]
I can confirm that the armv7 Linux binary works on an armv7 based router running OpenWRT.

While OpenSSL is not installed by default is can be done as as replacement for WolfSSL.

**Comment by ksingh-scogo** [comment]
@rapiz1  I am a FRP user , have been using it in prod since an year. But now i am stuck with FRP because of its large footprint ~10MB. 

For an IoT device we need a lightweight reverse proxy exactly like Rathole , because frp is too expensive (on flash storage).

I would be willing to try out things and help on this thread to make rathole work on MediaTek Devices running OpenWRT.  Where should i start and how can i contribute from testing point of view ? (i am not a rust developer) but interested to contribute on testing as i have access to a lot of OpenWRT based hardware.  Are you available on discord/IRC/Slack etc to work in real-time.

**Comment by rapiz1** [comment]
@ksingh-scogo you can start by picking up the right binary release. we already have releases for multiple embedded platforms. Then, you can try to package it for openwrt.

**Comment by fernvenue** [comment]
We agree that the OpenWrt ecosystem is too fragmented for us to take on and maintain a dedicated packaging effort right now, so we are closing this as not planned.

Thanks to everyone who weighed in over the years, including @ksingh-scogo in 2024 and others above. As noted in the thread, the existing embedded platform binaries already run on OpenWrt devices, so packaging into an .ipk is mostly a matter of building on top of that rather than a core code change.

If anyone is interested in picking this up, please feel free to reopen this issue or open a pull request at any time.
